File: client/src/main/java/ir/asparsa/hobbytaste/ui/list/holder/AboutUsViewHolder.java
package ir.asparsa.hobbytaste.ui.list.holder;

import android.os.Bundle;
import android.view.View;
import android.view.ViewGroup;
import butterknife.BindView;
import butterknife.ButterKnife;
import ir.asparsa.android.ui.fragment.recycler.BaseRecyclerFragment;
import ir.asparsa.android.ui.list.holder.BaseViewHolder;
import ir.asparsa.hobbytaste.R;
import ir.asparsa.hobbytaste.ui.list.data.AboutUsData;
import rx.functions.Action1;

/**
 * @author hadi
 * @since 3/16/2017 AD.
 */
public class AboutUsViewHolder extends BaseViewHolder<AboutUsData> {

    @BindView(R.id.about_us_layout)
    ViewGroup mLayout;

    public AboutUsViewHolder(
            View itemView,
            Action1<BaseRecyclerFragment.Event> observer,
            Bundle savedInstanceState
    ) {
        super(itemView, observer, savedInstanceState);
        ButterKnife.bind(this, itemView);
    }

    @Override public void onBindView(final AboutUsData data) {
        mLayout.setOnClickListener(new View.OnClickListener() {
            @Override public void onClick(View view) {
                mObserver.call(new AboutUsClick());
            }
        });
    }

    public static class AboutUsClick implements BaseRecyclerFragment.Event {
        AboutUsClick() {
        }
    }

}

File: client/src/main/kotlin/ir/asparsa/hobbytaste/ui/list/holder/AboutUsViewHolder.kt
package ir.asparsa.hobbytaste.ui.list.holder

import android.view.View
import android.view.ViewGroup
import butterknife.ButterKnife
import com.github.kittinunf.reactiveandroid.view.rx_click
import ir.asparsa.android.ui.list.holder.BaseViewHolder
import ir.asparsa.hobbytaste.R
import ir.asparsa.hobbytaste.ui.list.data.AboutUsData
import kotterknife.bindView
import rx.Observable

/**
 * @author hadi
 */
class AboutUsViewHolder(
        itemView: View
) : BaseViewHolder<AboutUsData>(itemView) {

    val mLayout: ViewGroup by bindView(R.id.about_us_layout)

    init {
        ButterKnife.bind(this, itemView)
    }

    fun clickStream(): Observable<AboutUsData> {
        return mLayout.rx_click().map { data }
    }
}
